engine/gui/src/gui.cpp

```diff
@@ -1472,7 +1472,7 @@ Result DeleteDynamicTexture(HScene scene, const dmhash_t texture_hash)
             if (opacity == 0.0f || n->m_Node.m_IsBone)
             {
                 entry.m_Node = INVALID_HANDLE;
-                entry.m_RenderKey = 0xFFFFFFFF;
```

Comment:
If we don't sort the invalid items to the end of the array, we'll accidentally keep old (invalid) nodes in the middle, and instead discard valid nodes.

Reply:
ouch (